lib/utils/date_utils.ts

```diff
@@ -112,10 +112,25 @@ function createDatePiece(year : number|null, month : number|null, day : number|n
     return date;
 }
 
-function createWeekDayDate(weekday : string, time : AbsoluteTime|null) {
-    const date = new Date;
+function weekdayToNumber(weekday : WeekDay) : number {
+    switch (weekday) {
+        case "monday": return 1;
+        case "tuesday": return 2;
+        case "wednesday": return 3;
+        case "thursday": return 4;
+        case "friday": return 5;
+        case "saturday": return 6;
+        case "sunday": return 7;
+    }
+    throw new Error(`Invalid weekday: ${weekday}`);
+}
 
-    // FIXME: implement this
+function createWeekDayDate(weekday : WeekDay, time : AbsoluteTime|null) {
+    const date = new Date;
+    // get the date of next specified weekday, today excluded
+    date.setDate(date.getDate() + (7 - date.getDay()) % 7 + weekdayToNumber(weekday));
```

Comment:
Is this right?
If today is a monday and we ask for next sunday, we first add 6 (7-1), and then add another 7 for sunday?

Reply:
ugh, yeah, take a look at the new version